atom/browser/api/atom_api_window.cc

```diff
@@ -80,6 +100,34 @@ void Window::OnPageTitleUpdated(bool* prevent_default,
   *prevent_default = Emit("page-title-updated", title);
 }
 
+void Window::OnSelectCertificate(
+    net::SSLCertRequestInfo* cert_request_info,
+    scoped_ptr<content::ClientCertificateDelegate> delegate) {
+  v8::Isolate* isolate = v8::Isolate::GetCurrent();
+  v8::Locker locker(isolate);
+  v8::HandleScope handle_scope(isolate);
+  std::vector<mate::Dictionary> cert_info_list;
+  certificate_delegate_.reset(delegate.get());
+
+  for (auto& cert : cert_request_info->client_certs) {
+    std::string encoded_data;
+    if (cert->valid_expiry().is_null() || cert->HasExpired()
+        || !net::X509Certificate::GetPEMEncoded(
+                cert->os_cert_handle(), &encoded_data))
+      continue;
+
+    mate::Dictionary info(isolate, v8::Object::New(isolate));
+    info.Set("data", encoded_data);
```

Comment:
We should store the index of the cert instead of copying all the data, a decent API is:

```js
app.on('select-certificate', function(host, certs, callback) {
   callback(certs[0]);
});
```

Of source `cert_request_info->client_certs` needs to be copied like the `delegate`.